---
title: 'Lab 5: Brassicaceae, Malvaceae, Ericaceae, and intro to phylogenetics'
author: Evan Hersh, Mannfred Boehm
date: "`r format(Sys.time(), '%d %B, %Y')`"
---

## Brassicaceae - Mustard family

**1.  Herbs.**  
**2.	Leaves simple, pinnatifid, or pinnately compound, alternate, without stipulates.**  
**3.	Flowers perfect, cruciform (cross or X shaped).  Sepals 4, free.  Petals 4, free, clawed.  Stamens 6 and tetradynamous (the 2 outer stamens shorter than the 4 inner stamens).  Pistil compound, ovary superior, carpels 2, placentation parietal.**  
**4.  Fruit a silique or silicle.**  
**5.  About 350 genera and 3,000 species, primarily North Temperate.**  

&nbsp;

![](media/diagram_Brassicaceae.png)

&nbsp;

Brassicaceae is the 4th largest family in B.C., with 169 species and varieties.  The family is very important economically, providing cabbage, broccoli, cauliflower (all species or varieties of *Brassica*), mustard, turnips, rapeseed or canola, radish (*Raphanus*), horseradish (*Amoracia rusticana*), and watercress (*Rorippa*).  Numerous genera are cultivated as ornamentals, including *Lobularia* (sweet alyssum), *Matthiola* (stocks), and *Aubrieta*.

&nbsp;

---

Brassicaceae flowers are very characteristic, making it easy to identify to family. The genera and species are more difficult however, and often require ripe fruit and sometimes a microscope to examine minute trichome structure. Let's look at some flowers. Below you can see the typical 'cruciform' (cross-like) structure of the petals, and 'tetradynamous' stamens.

<p align="center">
![](./media/Brassicaceae_1.jpeg){width=40%} ![](./media/Brassicaceae_2.jpeg){width=40%}
![](./media/Brassicaceae_3.jpeg){width=40%} ![](./media/Brassicaceae_4.jpeg){width=40%}
</p>

&nbsp;

The petals typically have a narrow "claw" at the base, with the "blade" bending outwards at the throat of the corolla. 

<p align="center">
![](./media/Brassicaceae_5.jpeg){width=40%} ![](./media/Brassicaceae_6.jpeg){width=40%}
</p>

&nbsp;

---

As mentioned above, much of the Brassicaceae key is concerned with the fruits. The family has two unique fruit types, siliques and silicles. Siliques are more the twice as long as they are wide (appearing long and narrow), while silicles are less than two times as long as wide (appearing short and wide in comparison). Can you identify which is which from the photos below?

<p align="center">
![](./media/Brassicaceae_7.jpeg){width=40%} ![](./media/Brassicaceae_8.jpeg){width=40%}
![](./media/Brassicaceae_9.jpeg){width=40%} ![](./media/Brassicaceae_10.jpeg){width=40%}
</p>

&nbsp;

Siliques and silicles are dehiscent fruit from a 2-carpellate, 2-loculed ovary. The two carpels (or valves) split away from a central structure called a "replum", which is a thin, often papery/translucent plane that sits in between the two valves like a wall. These fruit are typically flattened in two different ways: if the fruit is flattened parallel to the replum (so that the replum is rather wide) we refer to that as "compressed", and if the fruit is flattened perpendicular to the replum (so that the replum is quite thin) we call that "obcompressed". See if you can identify the two valves, replum, and the type of compression in the fruit below. 

<!-- (hint: it is also possible that the fruit are not flattened in either direction, which would be described as "terete", or cylindrical in cross-section...) -->

<p align="center">
![](./media/Brassicaceae_11.jpeg){width=40%} 
![](./media/Brassicaceae_12.jpeg){width=40%}
![](./media/Brassicaceae_13.jpeg){width=40%} ![](./media/Brassicaceae_14.jpeg){width=40%}
![](./media/Brassicaceae_15.jpeg){width=40%} ![](./media/Brassicaceae_16.jpeg){width=40%}
</p>

&nbsp;

---

Brassicaceae leaves can be simple or compound, and are arranged alternatively and without stipules. Many leaves are pinnately compound or "pinnatifid", which are pinnately divided or lobes but not all the way to the midrib of the leaf. 

<p align="center">
![](./media/Brassicaceae_17.jpeg){width=40%} 
</p>

&nbsp;

Another important characteristic for identification is trichomes, which can be quite complex and varied in the Brassicaceae. A stunning example of intricate trichome morphology is displayed in the genus *Draba* (Evan's personal favorite genus!!!), a hyper-diverse group that are typically only found in alpine areas. *Draba* (amongst many other members of Brassicaceae) have particular trichome patterns that can be useful (and sometimes necessary) to identify identify individual species. Can you identify different types of trichomes in the two *Draba* species below?

<p align="center">
![](./media/Brassicaceae_18.jpeg){width=40%}
![](./media/Brassicaceae_19.jpeg){width=40%}
</p>

&nbsp;

---

## Malvaceae - Mallow family

![](media/diagram_Malvaceae.png)

&nbsp;

---

## Ericaceae - Heath family

![](media/diagram_Ericaceae.png)

&nbsp;

---

## Introduction to phylogenetics

&nbsp;
&nbsp;

#### Motivation

Phylogenetics is one of the most important and controversial fields within plant taxonomy. The ability to sequence and compare plant DNA has split, grouped, and uncovered taxa in ways never expected. In modern plant taxonomy, phylogenetic relationships are a key piece of evidence to consider when delimiting species. In this lab, we'll be downloading and aligning DNA sequences from five species chosen from your iNaturalist observations. Using these sequences, we'll construct a phylogenetic tree and consider whether the species relationships support what you have learned about the taxonomic organization (e.g. grouping by family) of these taxa. 

---

&nbsp;
&nbsp;

#### Prerequisites 

Before starting this week's tutorial you will have: 

1. completed some pre-assignment reading on DNA barcoding, multiple sequence alignment, phylogenetics, and their role in plant taxonomy, 

2. completed some pre-assignment reading on why/how Genbank is used by plant taxonomists, 

3. chosen five plant species from your iNaturalist project to use in this activity.

---

&nbsp;
&nbsp;

#### Outcomes

By the end of this tutorial you will have: 

1. experience using Genbank, MUSCLE, and NCBI tree viewer.

2. familiarity with Genbank and the types of data that can be obtained, 

3. the tools and knowledge needed to create your own phylogenetic tree as part of a taxonomic research project.

---

&nbsp;
&nbsp;

#### Activity 1: Search Genbank for sequence data

After selecting your five focal species, [navigate to the Genbank website](https://www.ncbi.nlm.nih.gov/genbank/). From the drop-down menu next to the search bar, make sure that the database selected is *Nucleotide*. Type in scientific name of your first plant, followed by the name of the barcode. For demonstration purposes we'll use five species of pussytoes (*Antennaria*, Asteraceae), and the rbcL barcode. Here's what our first search looks like:

![](media/lab5_genbank1.png) 

The search will return list of sequence submissions from various authors and projects that have barcoded *Antennaria*. When possible, use sequences that were generated from the same project &mdash; you can check by clicking on an entry and looking at the AUTHORS and TITLE entries. For our *Antennaria* sequences, we'll be using rbcL sequences generated by [Fenneman and Graham (2016)](https://open.library.ubc.ca/cIRcle/collections/ubctheses/24/items/1.0340546) as part of their contribution to barcode over 5000 vascular plants [representing the flora of Canada](https://bsapubs.onlinelibrary.wiley.com/doi/full/10.3732/apps.1700079). 

After entering your search query, click on the 'FASTA' option under any entry to see the DNA sequence:

![](media/lab5_genbank2.png)

To save the sequence, we'll copy sequence string, as well as the header information (everything in the box above). Paste the FASTA entry into a text editor. Repeat this procedure for the remaining four taxa, pasting the FASTA entries into your text document, with each entry separated by a space. Then, simplify the headers by writing the scientific name with the genus and species separated by an underscore "_". Make sure to keep the ">" symbol at the beginning. 

![](media/lab5_genbank3.png)

&nbsp;
&nbsp;

#### Activity 2: Multiple Sequence Alignment

In order to compare the similarities of our sequences, we need to first align them. This ensures that any differences between samples are not caused simply by different reading frames. While we *could* align our sequences by eye, we'll use the MUSCLE algorithm which is much faster and reproducible. [Navigate to the MUSCLE webpage](https://www.ebi.ac.uk/Tools/msa/muscle/), and paste all of your sequencs into the first box:

![](media/lab5_muscle1.png)

After pasting your sequences, scroll down and click "Submit". The results page will show you the optimal alignment of your five rbcL sequences. To see the associated phylogram, click on "Phylogenetic Tree" tab:

![](media/lab5_muscle2.png)

**MB** maybe we want to get them to continue by pasting the Tree Data in to [NCBI Tree Viewer?](https://www.ncbi.nlm.nih.gov/projects/treeview/)

