<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mannfred Boehm" />


<title>R Assignment 4: Creating Phylogenetic Trees using Genbank data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOL 324 Labs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="BIOL_324_Lab_2.html">Lab 2</a>
</li>
<li>
  <a href="BIOL_324_Lab_6.html">Lab 6</a>
</li>
<li>
  <a href="BIOL_324_Lab_7.html">Lab 7</a>
</li>
<li>
  <a href="BIOL_324_Lab_8.html">Lab 8</a>
</li>
<li>
  <a href="BIOL_324_Lab_9.html">Lab 9</a>
</li>
<li>
  <a href="BIOL_324_Lab_10.html">Lab 10</a>
</li>
<li>
  <a href="LinneanTypification.html">Linnean Typification</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Assignment 4: Creating Phylogenetic Trees using Genbank data</h1>
<h4 class="author">Mannfred Boehm</h4>
<h4 class="date">17 August, 2020</h4>

</div>


<p><br> <br></p>
<div id="motivation" class="section level4">
<h4>Motivation</h4>
<p>Phylogenetics is one of the most important and controversial fields within plant taxonomy. The ability to sequence and compare plant DNA has split, grouped, and uncovered taxa in ways never expected. In modern plant taxonomy, phylogenetic relationships are a key piece of evidence to consider when delimiting species. In this lab, we’ll be downloading and aligning DNA sequences from five species of <em>Opuntia</em>. Using these sequences, we’ll construct a phylogenetic tree and consider how the species relationships hold up to the published literature.</p>
<hr />
<p><br> <br></p>
</div>
<div id="prerequisites" class="section level4">
<h4>Prerequisites</h4>
<p>Before starting this week’s tutorial you will have: <br></p>
<ol style="list-style-type: decimal">
<li><p>completed some pre-assignment reading on DNA barcoding, multiple sequence alignment, phylogenetics, and their role in plant taxonomy, <br></p></li>
<li><p>completed some pre-assignment reading on why/how Genbank is used by plant taxonomists, <br></p></li>
<li><p>navigated through the Genbank website and understand what it is and why it exists.</p></li>
<li><p><strong>MB</strong> Get students to read (for example, something like) <a href="https://bsapubs.onlinelibrary.wiley.com/doi/full/10.3732/ajb.1100375">Majure et al (2012), Am. J. Bot.</a> if we end up using <em>Opuntia</em> as an example…</p></li>
</ol>
<hr />
<p><br> <br></p>
</div>
<div id="outcomes" class="section level4">
<h4>Outcomes</h4>
<p>By the end of this tutorial you will have:</p>
<ol style="list-style-type: decimal">
<li><p>experience using Genbank, CLUSTAL, and the most common phylogenetics packages in R</p></li>
<li><p>familiarity with Genbank and the types of data that can be obtained,</p></li>
<li><p>the tools and knowledge needed to create your own phylogenetic tree as part of a taxonomic research project,</p></li>
</ol>
<p>   </p>
<hr />
</div>
<div id="activity-1-downloading-aligning-and-graphing-dna-sequences-from-genbank" class="section level4">
<h4>Activity 1: Downloading, Aligning, and Graphing DNA Sequences from Genbank</h4>
<p>In previous Assignments we familiarized ourselves with GBIF, R, and the R packages that allow us to connect to databases. Today, we’ll learn how to use our newfound R skills to download and align nucleotide sequences from <a href="https://www.ncbi.nlm.nih.gov/nuccore">GenBank</a>, a repository of DNA sequences hosted by the U.S. National Center for Biotechnology Information (NCBI). We’ll then use the aligned sequences to create a phylogenetic tree that will organize species based on their relatedness.</p>
<p>In this lab, we’ll explore further the genus <em>Opuntia</em>:</p>
<p><em>Opuntia basilaris</em> (Mojave Desert, Colorado, Utah) <em>Opuntia fragilis</em> (Northwestern) <em>Opuntia humifusa</em> (Eastern) <em>Opuntia polyacantha</em> (Great Plains, foothills of Rocky Mountains) <em>Opuntia stricta</em> (Gulf Coast and Caribbean)</p>
<p><strong>MB</strong> Could add in some natural history background to better motivate this activity. <br></p>
<p>As usual, we’ll start by installing the necessary packages using <code>install.packages()</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="kw">c</span>(<span class="st">&#39;ape&#39;</span>, <span class="st">&#39;msa&#39;</span>, <span class="st">&#39;seqinr&#39;</span>))</span></code></pre></div>
<p>Then we’ll attach some packages designed specifically for connecting with the Genbank database (<code>ape</code>), filtering and organizing our data (<code>tidyverse</code>), aligning sequences (<code>msa</code>), concatenating sequences (<code>seqinr</code>), and constructing phylogenetic trees (<code>ape</code>):</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># attach packages once they are installed </span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw">library</span>(ape)      </span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="kw">library</span>(here)   </span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="kw">library</span>(msa)    </span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="kw">library</span>(seqinr) </span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
<p>First, we’re going to build a spreadsheet that will help us keep track of our <em>Opuntia</em> species and the GenBank accession codes for both the matK and rbcL genes for each species. The accession numbers were found by searching the GenBank website like so:</p>
<p><img src="media/genbank.jpg" /></p>
<p>Once the accession numbers are found, they can be saved in a data frame:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># build data frame</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co"># accession numbers found by searching GenBank</span></span>
<span id="cb3-3"><a href="#cb3-3"></a>species &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;O. basilaris&#39;</span>, <span class="st">&#39;O. fragilis&#39;</span>, <span class="st">&#39;O. humifusa&#39;</span>, <span class="st">&#39;O. polyacantha&#39;</span>, <span class="st">&#39;O. stricta&#39;</span>) </span>
<span id="cb3-4"><a href="#cb3-4"></a>matK_acc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;JF786750.1&#39;</span>, <span class="st">&#39;JF786786.1&#39;</span>, <span class="st">&#39;JF786791.1&#39;</span>, <span class="st">&#39;JF786823.1&#39;</span>, <span class="st">&#39;JF786853.1&#39;</span>) </span>
<span id="cb3-5"><a href="#cb3-5"></a>rbcL_acc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;JF787189.1&#39;</span>, <span class="st">&#39;JF787223.1&#39;</span>, <span class="st">&#39;JF787227.1&#39;</span>, <span class="st">&#39;JF787259.1&#39;</span>, <span class="st">&#39;JF787288.1&#39;</span> ) </span>
<span id="cb3-6"><a href="#cb3-6"></a></span>
<span id="cb3-7"><a href="#cb3-7"></a>df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(species, matK_acc, rbcL_acc)</span></code></pre></div>
<p>Now we’re ready to search and download the matK sequence for each <em>Opuntia</em> species, using the <code>read.Genbank()</code> function:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># call GenBank</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>matK &lt;-<span class="st"> </span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="st">  </span><span class="kw">read.GenBank</span>(</span>
<span id="cb4-4"><a href="#cb4-4"></a>    df<span class="op">$</span>matK_acc,</span>
<span id="cb4-5"><a href="#cb4-5"></a>    <span class="dt">species.names =</span> <span class="ot">FALSE</span>,</span>
<span id="cb4-6"><a href="#cb4-6"></a>    <span class="dt">as.character =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="st">  </span><span class="kw">lapply</span>(., paste0,<span class="dt">collapse=</span><span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span><span class="co"># remove spaces</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="st">  </span><span class="kw">lapply</span>(., str_to_upper) <span class="co"># capitalize </span></span>
<span id="cb4-9"><a href="#cb4-9"></a></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co"># replace Accession IDs with English names</span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="kw">names</span>(matK) &lt;-<span class="st"> </span>df<span class="op">$</span>species</span></code></pre></div>
<p>We can inspect the DNA sequence for <em>O. basilaris</em> by running:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">head</span>(matK<span class="op">$</span><span class="st">`</span><span class="dt">O. basilaris</span><span class="st">`</span>)</span></code></pre></div>
<pre><code>## [1] &quot;TCAATATTTCCCTTTTTCGAGGACAAATTCTTACATTTAAATTATGTGTTAAAAATATTAATACCTTACCCCATTCATCTAGAAATCTTGGTTCAAACTCTTCGTTACTGGGTGAAAGATGCTTCTTCTTTGCATTTATTACGATTCTTTCTTTATGAGTATCGTAATTGGAATAGTCTTATTACTCCCCAAAAATCCATTTCTATTTTTTCAAAAAAGAATCAACGATTATTCTTGTTCCTATATAATTTCCATGTATGTGAATACGAATCCATTTTCGTTTTTCTCTGTAACCAATCCTCTCATTTACGATCTTTTACTTTTGGAGCCCTTCTTGAACGAATCTATTTTTATGGAAAGCGTGAATATCTAGTAAAAGTAAAAACTTTATCTAAGGATTTTCGTCTTATCTTATGGCTTTTCAAAGACCCTTTCCTGCATTATGAATTGTATCGAGGAAAAGAAATTCTGGCTTCAAAGGGGACATCTCTTCTGATGCATAAATGGAAATATTATCTTAACAATTTTTGGCAATGCCATTTTTCCCTGTGGTCTCAACCAAGAAGAATCTATATCAATCGATTATCAAAGCATTCTCTCGACTTTATGAGTTTTTTTTCAAGTGTTCGACTCAATTCTTCAGTGGTACGGAGTCAAATGGTAGAAAATTCATTTCTAATAGATAATCCTATTAAGAAATTCGATACCATAGTTCGAATTATTCCTCTGGTTGGATCGTTGGCTAAAGCGAAATTTTGTAACGTATTAGGACATCCCGTTAGTAAGTCGGTCTGGACCGATTTATTAGATTCTGATATTATTGATCGGTTTGGGCGCATATGCAGAAATCTTTCTCATTATTATAGTGGCTCTTCAAGAAAAAAGAGTTTGTATCGAATAAAGTATATA&quot;</code></pre>
<p>Cool, <a href="https://pubmed.ncbi.nlm.nih.gov/16100764/">it’s like we’re on CSI</a>. The code for searching and downloading the rbcL sequences is similar:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># call GenBank</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>rbcL &lt;-<span class="st"> </span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="st">  </span><span class="kw">read.GenBank</span>(</span>
<span id="cb7-4"><a href="#cb7-4"></a>    df<span class="op">$</span>rbcL_acc,</span>
<span id="cb7-5"><a href="#cb7-5"></a>    <span class="dt">species.names =</span> <span class="ot">FALSE</span>,</span>
<span id="cb7-6"><a href="#cb7-6"></a>    <span class="dt">as.character =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="st">  </span><span class="kw">lapply</span>(., paste0,<span class="dt">collapse=</span><span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span><span class="co"># remove spaces</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="st">  </span><span class="kw">lapply</span>(., str_to_upper) <span class="co"># capitalize </span></span>
<span id="cb7-9"><a href="#cb7-9"></a></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co"># replace Accession IDs with English names</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="kw">names</span>(rbcL) &lt;-<span class="st"> </span>df<span class="op">$</span>species</span></code></pre></div>
<p>We now have two DNA sequence objects, <code>matK</code> and <code>rbcL</code>, that contain sequences from each of our five <em>Opuntia</em> species. In the next step, we’ll use the <code>msa()</code> function to align our five matK sequences so that their ends are lined up. You could also do this by eye, but it’s a bit tedious and not as reproducible:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># matK alignment</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>matK_alignment &lt;-<span class="st"> </span></span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="st">  </span><span class="kw">msa</span>(<span class="kw">unlist</span>(matK),</span>
<span id="cb8-4"><a href="#cb8-4"></a>      <span class="dt">type =</span><span class="st">&#39;dna&#39;</span>, </span>
<span id="cb8-5"><a href="#cb8-5"></a>      <span class="dt">method =</span><span class="st">&#39;ClustalW&#39;</span>,</span>
<span id="cb8-6"><a href="#cb8-6"></a>      <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># inspect alignment</span></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw">print</span>(matK_alignment) </span></code></pre></div>
<pre><code>## CLUSTAL 2.1  
## 
## Call:
##    msa(unlist(matK), type = &quot;dna&quot;, method = &quot;ClustalW&quot;, verbose = TRUE)
## 
## MsaDNAMultipleAlignment with 5 rows and 911 columns
##     aln                                                    names
## [1] ----CATATT-CCCTTTT-CGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. fragilis
## [2] ----CATATTTCCCTTTT-CGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. humifusa
## [3] CATCAATATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. polyacantha
## [4] ---TCAAATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. stricta
## [5] --TCAATATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. basilaris
## Con ---?CATATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA Consensus</code></pre>
<p>By using the <code>print()</code> function, we see that our sequences are aligned pretty well! You can also notice some differences in nucleotide position between the species. Those are ‘single nucleotide polymorphisms’ (SNPs) that, when considered across the entire genome, distinguish one species from another.</p>
<p>We’ll now align our rbcL sequences:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>rbcL_alignment &lt;-<span class="st"> </span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="st">  </span><span class="kw">msa</span>(<span class="kw">unlist</span>(rbcL),</span>
<span id="cb11-3"><a href="#cb11-3"></a>      <span class="dt">type =</span><span class="st">&#39;dna&#39;</span>, </span>
<span id="cb11-4"><a href="#cb11-4"></a>      <span class="dt">method =</span><span class="st">&#39;ClustalW&#39;</span>,</span>
<span id="cb11-5"><a href="#cb11-5"></a>      <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## use default substitution matrix</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># inspect alignment</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="kw">print</span>(rbcL_alignment)</span></code></pre></div>
<pre><code>## CLUSTAL 2.1  
## 
## Call:
##    msa(unlist(rbcL), type = &quot;dna&quot;, method = &quot;ClustalW&quot;, verbose = TRUE)
## 
## MsaDNAMultipleAlignment with 5 rows and 807 columns
##     aln                                                    names
## [1] TTTTATAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAATTAAGAG- O. basilaris
## [2] --------------------------...------------------------- O. stricta
## [3] ---------------TATGTCGAAAT...------------------------- O. humifusa
## [4] -----TAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAAT------- O. fragilis
## [5] --TTTTAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAATTAAGTTG O. polyacantha
## Con -----TAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAAT------- Consensus</code></pre>
<p>We’re ready to create some phylogenetic trees! First, we’ll convert our alignment objects to <code>seqinr</code> objects. Then, we’ll use the <code>dist.alignment()</code> function to calculate how similar or different each species’ sequence is from one another. We’ll do this for the matK, rbcL, and concatenated sequences, so that we can compare the trees:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="co"># convert alignment object into seqinr object </span></span>
<span id="cb15-2"><a href="#cb15-2"></a>matK_r &lt;-<span class="st"> </span><span class="kw">msaConvert</span>(matK_alignment, <span class="dt">type=</span><span class="st">&quot;seqinr::alignment&quot;</span>) </span>
<span id="cb15-3"><a href="#cb15-3"></a>rbcL_r &lt;-<span class="st"> </span><span class="kw">msaConvert</span>(rbcL_alignment, <span class="dt">type=</span><span class="st">&quot;seqinr::alignment&quot;</span>) </span>
<span id="cb15-4"><a href="#cb15-4"></a>  </span>
<span id="cb15-5"><a href="#cb15-5"></a><span class="co"># create distance matrices</span></span>
<span id="cb15-6"><a href="#cb15-6"></a>matK_dist &lt;-<span class="st"> </span><span class="kw">dist.alignment</span>(matK_r)</span>
<span id="cb15-7"><a href="#cb15-7"></a>rbcL_dist &lt;-<span class="st"> </span><span class="kw">dist.alignment</span>(rbcL_r)</span></code></pre></div>
<p>Now, we’ll use the <code>nj()</code> function to construct trees. In the next step, we’ll ‘root’ our trees by deciding which species is the outgroup:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># neighbour joining trees</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>matK_tree &lt;-<span class="st"> </span><span class="kw">nj</span>(matK_dist)</span>
<span id="cb16-3"><a href="#cb16-3"></a>rbcL_tree &lt;-<span class="st"> </span><span class="kw">nj</span>(rbcL_dist)</span>
<span id="cb16-4"><a href="#cb16-4"></a></span>
<span id="cb16-5"><a href="#cb16-5"></a>rbcL_rooted &lt;-<span class="st"> </span><span class="kw">root</span>(rbcL_tree, <span class="dt">outgroup =</span> <span class="st">&#39;O. basilaris&#39;</span>, <span class="dt">resolve.root =</span> T)</span>
<span id="cb16-6"><a href="#cb16-6"></a>matK_rooted &lt;-<span class="st"> </span><span class="kw">root</span>(matK_tree, <span class="dt">outgroup =</span> <span class="st">&#39;O. basilaris&#39;</span>, <span class="dt">resolve.root =</span> T)</span></code></pre></div>
<p>Plot your new trees:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="co"># plot trees</span></span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="kw">plot</span>(matK_rooted, <span class="dt">main =</span><span class="st">&#39;matK tree&#39;</span>)</span></code></pre></div>
<p><img src="BIOL_324_Lab_9_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="kw">plot</span>(rbcL_rooted, <span class="dt">main =</span> <span class="st">&#39;rbcL tree&#39;</span>)</span></code></pre></div>
<p><img src="BIOL_324_Lab_9_files/figure-html/unnamed-chunk-13-2.png" width="672" /></p>
<p>Finally, we’ll save our trees so that we can use them again in Lab 10:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="kw">saveRDS</span>(matK_rooted, <span class="dt">file =</span> <span class="kw">here</span>(<span class="st">&#39;R_scripts/matK_rooted.rds&#39;</span>))</span>
<span id="cb19-2"><a href="#cb19-2"></a><span class="kw">saveRDS</span>(rbcL_rooted, <span class="dt">file =</span> <span class="kw">here</span>(<span class="st">&#39;R_scripts/rbcL_rooted.rds&#39;</span>))</span></code></pre></div>
<p><strong>-KNOWLEDGE CHECK-</strong> How do the matK and rbcL trees differ? Which tree best aligns with the <a href="https://bsapubs.onlinelibrary.wiley.com/doi/full/10.3732/ajb.1100375">Majure tree (Figure 3)</a>? According to Majure et al (2012), why might our trees differ?</p>
<!-- For consensus trees: -->
<!-- Recall that matK and rbcL are two separate segments of the chloroplast genome. By sticking them together, we'll create a super-sequence that contains more information than either of the sequences alone. The more sequence information, the better, because it gives us a bigger sample of the entire genome. Concatenation is done by the `seqinr` package, so we'll need to convert our alignment objects into `seqinr` objects: -->
<!-- ```{r, results = 'hide'} -->
<!-- # convert alignment object into seqinr object -->
<!-- matK_r <- msaConvert(matK_alignment, type="seqinr::alignment") -->
<!-- rbcL_r <- msaConvert(rbcL_alignment, type="seqinr::alignment") -->
<!-- # concatenate (merge) sequences -->
<!-- conseq <- mapply(paste0, matK_r$seq, rbcL_r$seq) -->
<!-- names(conseq) <- df$species -->
<!-- # re-align -->
<!-- con_alignment <- -->
<!--   msa(unlist(conseq), -->
<!--       type ='dna',  -->
<!--       method ='ClustalW', -->
<!--       verbose = TRUE) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # inspect  -->
<!-- print(con_alignment) -->
<!-- # convert alignment object into seqinr object -->
<!-- con_r <- msaConvert(con_alignment, type="seqinr::alignment") -->
<!-- ``` -->
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
