<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Lily-like monocots, Iridaceae, Orchidaceae, and Phylogenetics in R</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOL 324 Labs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Syllabus.html">Syllabus</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Resources
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Morphological_Terminology.html">Morphological Terminology</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="BIOL_324_Project_1.html">Project 1 (iNaturalist)</a>
    </li>
    <li>
      <a href="LinneanTypification.html">Linnean Typification</a>
    </li>
    <li>
      <a href="BIOL_324_Project_2.html">Project 2 (Phylogeography)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Labs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="BIOL_324_Lab_1.html">Lab 1</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_2.html">Lab 2</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_3.html">Lab 3</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_4.html">Lab 4</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_5.html">Lab 5</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_6.html">Lab 6</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_7.html">Lab 7</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_8.html">Lab 8</a>
    </li>
    <li>
      <a href="BIOL_324_Lab_9.html">Lab 9</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Lily-like monocots, Iridaceae, Orchidaceae, and Phylogenetics in R</h1>

</div>


<div id="lily-like-monocots" class="section level1">
<h1>Lily-like Monocots</h1>
<hr />
<p><strong>1. Usually herbs.</strong><br />
<strong>2. Stems often modified as rhizomes, bulbs, or corms. Leaves often linear or strap shaped, simple, usually alternate, but may be basal or whorled, without stipules.</strong><br />
<strong>3. Flowers usually perfect, actinomorphic, tepals 6, free or partially fused, or sometimes a hypanthium present. Stamens 6 (rarely 3). Pistil compound, ovary superior or inferior (those whith umbellate inflorescences and inferior ovaries are often put in the Amaryllidaceae), carpels 3, placentation axile.</strong><br />
<strong>4. Fruit a capsule or berry.</strong><br />
<strong>5. About 240 genera and 4,200 species, widely distributed.</strong></p>
<p>Traditionally the bulk of the lily-like petaloid monocots were divided into two main families Liliaceae <em>sensu lato</em> (lily family mostly with superior ovary) and Amaryllidaceae <em>s.l.</em> (daffodil family mostly with inferior ovary). Cronquist took the approach of having a very broadly defined Liliaceae, which included both these families as well as others. The APG found that plants belonging to Cronquist’s Liliaceae were highly heterogeneous (a circumstance variously known as a “rag-bag” or “taxodump”). The components of this broadly circumscribed Liliaceae <em>sensu latissimo</em> are assignable to both the APG orders Asparagales and Liliales. The old Liliaceae has therefore been broken up and lily-like plants have been assigned to a number of smaller monophyletic families.</p>
<p><strong>LILIALES</strong> <em>sensu</em> APG Comprises:<br />
Alstroemeriaceae, Campynemataceae, Colchicaceae (autumn crocus family), Corsiaceae, Liliaceae (lily family), Luzuriagaceae, Melanthiaceae (trillium family), Petermanniaceae, Philesiaceae, Rhipogonaceae, Smilacaceae (smilax family).</p>
<p><strong>ASPARAGALES</strong> <em>sensu</em> APG Comprises:<br />
Agapanthaceae (Alliaceae <em>s.l.</em>), Agavaceae (agave family), Alliaceae (onion family), Amaryllidaceae (daffodil family) (Alliaceae <em>s.l.</em>), Aphyllanthaceae, Asparagaceae (asparagus family), Asphodelaceae, Asteliaceae, Blandfordiaceae, Boryaceae, Doryanthaceae, Hemerocallidaceae (day lily family), Hyacinthaceae (hyacinth family), Hypoxidaceae, Ixioliriaceae, Lanariaceae, Laxmanniaceae, Ruscaceae (butcher’s broom family), Tecophilaeaceae, Themidaceae, Xanthorrhoeaceae, Xeronemaceae, Iridaceae (irises), Orchidaceae (orchids).</p>
<p>It should be noted that the Agapanthaceae (<em>Agapanthus</em> family), the Alliaceae (onion family) and the Amaryllidaceae (daffodil family) are all closely related and may be united in a single family (Alliaceae <em>s.l.</em>)</p>
<hr />
<div id="diagrams" class="section level2">
<h2>Diagrams</h2>
<p><img src="media/diagram_lily1.png" style="width:100.0%" /></p>
<p><br></p>
<p><img src="media/diagram_lily2.png" style="width:100.0%" /></p>
<p><br></p>
<p><img src="media/diagram_lily3.png" style="width:100.0%" /></p>
<hr />
</div>
<div id="leaves" class="section level2">
<h2>Leaves</h2>
<p>Leaves in these families tend to be linear or strap-shaped, simple, entire, alternate (but sometimes basal or whorled), and with parallel leaf venation. This is a good search image for all of the families you’ll be learning in this lab (lily-like monocots, Iridaceae, and Orchidaceae), and monocots in general. Here are some examples from members of the lily-like monocots:</p>
<p align="center">
<img src="media/Lily1.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Lily2.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Lily3.jpeg" style="width:100.0%" />
</p>
<hr />
<p>In many cases, the stems are modified into underground bulbs, corms, or rhizomes. You are probably familiar with the genus <em>Allium</em>, which has been cultivated extensively by humans and includes onion, garlic, scallion, shallot, leek, and chives. These species (and many others in the Amaryllidaceae) have modified leaves and stems. The stem is modified down to a little nub or plate at the center/base of the bulb, and is surrounded by fleshy scale-like leaves (picture the layers of an onion). The less-modified leaves that emerge from the bulb are long, hollow, and cylindrical in shape.</p>
<p align="center">
<img src="media/Lily4.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Lily5.jpg" style="width:100.0%" />
</p>
<p><br></p>
<p>Bulbs can also be found in the Asparagaceae family, as seen in the “pregnant onion” (<em>Ornithogalum caudatum</em>). In this case, the majority of the bulb sits above-ground, unlike what we typically see in <em>Allium</em>.</p>
<p align="center">
<img src="media/Lily6.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="flowers" class="section level2">
<h2>Flowers</h2>
<p>Species in the genus <em>Lilium</em> (which can be bought as ornamentals from most florists) are nice and big, and do a great job of displaying the typical flower structure of lily-like monocots. Note the undifferentiated sepal and petal whorls (thus we call them tepals) and unfused parts in 3’s. How many carpels do you think make up the ovary of this flower? Ovaries can either be superior or inferior depending on the family.</p>
<p align="center">
<img src="media/Lily7.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Lily8.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p>Here’s a cross-section of the ovary:</p>
<p align="center">
<img src="media/Lily9.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p>You may have noticed the green patches near the base of each petal. Those are actually the nectaries! Taking a closer look, you can see the glistening nectar being exuded along these organs.</p>
<p align="center">
<img src="media/Lily10.jpeg" style="width:49.0%" /> <img src="media/Lily11.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<p>Here are more examples of beautiful, showy, and fancy lily-like monocots :-). Notice how they all have parts in 3’s and are generally actinomorphic.</p>
<em>Clintonia uniflora</em> (Liliaceae)
<p align="center">
<img src="media/Lily12.jpg" style="width:100.0%" />
</p>
<p><br></p>
<em>Lilium columbianum</em> (Liliaceae)
<p align="center">
<img src="media/Lily13.jpg" style="width:49.0%" /> <img src="media/Lily14.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<p>Here you can see the beautiful wild daffodil, which has a trumpet-shaped corolla appendage called a “corona”. All species in the genus contain a poisonous alkaloid called lycorine.</p>
<em>Narcissus pseudonarcissus</em> (Amaryllidaceae)
<p align="center">
<img src="media/Lily15.jpg" style="width:100.0%" />
</p>
<p><br></p>
<em>Calochortus</em> sp. (Liliaceae)
<p align="center">
<img src="media/Lily16.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<em>Tricyrtis</em> sp. (Liliaceae)
<p align="center">
<img src="media/Lily17.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="inflorescences" class="section level2">
<h2>Inflorescences</h2>
<p>Many species in the Liliaceae have a simple raceme inflorescence:</p>
<p align="center">
<img src="media/Lily18.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p>In Amaryllidaceae and Alliaceae (the onion family), however, we often find umbel or pseudo-umbel inflorescences similar to what we saw in Apiaceae. These inflorescences are often subtended by a “spathe” bract, which enclose the flowers while they are developing.</p>
<em>Allium</em> sp. (Alliaceae)
<p align="center">
<img src="media/Lily19.jpeg" style="width:49.0%" /> <img src="media/Lily22.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<em>Allium</em> sp. (Alliaceae)
<p align="center">
<img src="media/Lily20.jpeg" style="width:49.0%" /><img src="media/Lily21.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<em>Agapanthus</em> sp. (Amaryllidaceae)
<p align="center">
<img src="media/Lily23.jpeg" style="width:49.0%" /><img src="media/Lily24.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="fruit" class="section level2">
<h2>Fruit</h2>
<p>Fruit in these families are typically capsules or berries. Here, you can see a dehisced <em>Lilium</em> capsule (notice the cool teeth in between the ovary segments) and winged seeds.</p>
<p align="center">
<img src="media/Lily25.jpeg" style="width:49.0%" /><img src="media/Lily26.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<p>Here you can see the beautiful berries of False Solomon’s seal (<em>Maianthemum racemosum</em>).</p>
<p align="center">
<img src="media/Lily27.jpeg" style="width:49.0%" /><img src="media/Lily28.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="knowledge-check-9.1" class="section level2">
<h2>Knowledge Check 9.1</h2>
<p>Draw any of the lily-like monocots pictured above (or your favourite species if you have one ;-)) and label all visible flower parts. Take note of the number of parts in each whorl, the inflorescence type, and whether or not a spathe bract is present.</p>
<hr />
</div>
</div>
<div id="iridaceae" class="section level1">
<h1>Iridaceae</h1>
<p><strong>The Iris family</strong></p>
<hr />
<p><strong>1. Herbs</strong><br />
<strong>2. Underground stems modified as rhizomes, bulbs, or corms. Leaves simple, mostly basal and linear, usually equitant.</strong><br />
<strong>3. Flowers perfect, actinomorphic or zygomorphic, subtended by a spathe like bract. Tepals 6, free or partially fused. Stamens 3. Pistil compound, ovary inferior, carpels 3, placentation usually axile.</strong><br />
<strong>4. Fruit a capsule.</strong><br />
<strong>5. About 60 genera and 1,500 species, widely distributed.</strong></p>
<p>Only 2 genera, <em>Sisyrinchium</em> and <em>Iris</em>, are found in B.C. Many genera are cultivated as ornamentals, including <em>Crocus</em>, <em>Iris</em>, <em>Freesia</em>, and <em>Gladiolus</em>.</p>
<hr />
<div id="diagrams-1" class="section level2">
<h2>Diagrams</h2>
<p><img src="media/diagram_Iridaceae1.png" style="width:100.0%" /></p>
<p><br></p>
<p><img src="media/diagram_Iridaceae2.png" style="width:100.0%" /></p>
<hr />
</div>
<div id="leaves-1" class="section level2">
<h2>Leaves</h2>
<p>Similar to the lily families, members of the Iridaceae also have simple, linear, and entire leaves with parallel venation and stems that are modified into underground corms or rhizomes. However, leaves in Iridaceae tend to be “equitant”, which means that they are straddling or overlapping each other. This is a reliable character to help you differentiate the Iridaceae from the lily-like monocots. Here are some examples:</p>
<p align="center">
<img src="media/Iridaceae1.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Iridaceae2.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Iridaceae3.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Iridaceae4.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="flowers-1" class="section level2">
<h2>Flowers</h2>
<p>Also similar to the lily families, the flowers in Iridaceae tend to be large and beautiful with parts in 3’s. The ovary is always inferior in Iridaceae, and the flowers are typically subtended by a conspicuous spathe bract. Flowers can either be actinomorphic or zygomorphic.</p>
<p>Here you can see the parts in 3’s and inferior ovary of <em>Iris domestica</em>.</p>
<p align="center">
<img src="media/Iridaceae5.jpeg" style="width:49.0%" /><img src="media/Iridaceae6.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<p>Many species of <em>Iris</em> (including the ones whose data you analyzed for your first R exercise in <a href="BIOL_324_Lab_6.Rmd">Lab 6</a>) have a rather unique and modified flower structure. The sepals and petals are differentiated (so no longer tepals), and the styles are petaloid as well. The stamens are actually hidden between the petaloid styles and the sepals.</p>
<em>Iris sibirica</em>
<p align="center">
<img src="media/Iridaceae10.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Iridaceae11.jpeg" style="width:49.0%" /> <img src="media/Iridaceae12.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<p>You can see the sneaky stamen quite well in this photo:</p>
<em>Iris tectorum</em>
<p align="center">
<img src="media/Iridaceae13.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p>Here you can see the large, beautiful zygomorphic flowers of <em>Gladiolus</em>. Wild species of <em>Gladiolus</em> actually have tiny flowers, so the ornamental varieties shown below are the product of centuries of artificial selection and hybridization.</p>
<p align="center">
<img src="media/Iridaceae7.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Iridaceae8.jpeg" style="width:49.0%" /><img src="media/Iridaceae9.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<p>Here you can see <em>Sisyrinchium idahoense</em> (Idaho blue-eyed grass), which can be found in moist grassy meadows in B.C.</p>
<p align="center">
<img src="media/Iridaceae14.jpg" style="width:100.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="knowledge-check-9.2" class="section level2">
<h2>KNOWLEDGE CHECK 9.2</h2>
<p>Make a drawing of a typical <em>Iris</em> flower, and make sure to label the following:</p>
<ul>
<li>petals</li>
<li>sepals</li>
<li>petaloid style branches</li>
<li>stamens</li>
<li>ovary</li>
<li>spathe bract</li>
</ul>
<hr />
<hr />
</div>
</div>
<div id="orchidaceae" class="section level1">
<h1>Orchidaceae</h1>
<p><strong>The Orchid family</strong></p>
<hr />
<p><strong>1. Herbs, sometimes saprophytic.</strong><br />
<strong>2. Leaves simple, usually alternate, with a sheathing base.</strong><br />
<strong>3. Flowers perfect, zygomorphic. Sepals 3, free. petals 3, free; one modified as the lip (labellum). Fertile stamen 1 (rarely 2). Pistil compound, ovary inferior, carpels 3, placentation parietal. The styles, stigma, and stamens fused to form the column.</strong><br />
<strong>4. Fruit a capsule.</strong><br />
<strong>5. Probably the second largest family of angiosperms, with about 450 genera and 20,000 species, widely distributed but mainly tropical.</strong></p>
<p>The Orchidaceae in B.C. are terrestrial, but most tropical species are epiphytes (growing on top of other plants). Some of ours lack chlorophyll and are saprophytic. Many genera are cultivated indoors for their flowers. Vanilla is derived from the fermented fruits of the genus <em>Vanilla</em>.</p>
<hr />
<div id="diagrams-2" class="section level2">
<h2>Diagrams</h2>
<p><img src="media/diagram_Orchidaceae1.png" style="width:100.0%" /></p>
<hr />
</div>
<div id="leaves-2" class="section level2">
<h2>Leaves</h2>
<p>Orchid leaves are pretty uninspiring in comparison to their spectacular flowers. Similar to other monocots, the leaves are simple and entire with parallel veins.</p>
<p align="center">
<img src="media/Orchidaceae1.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Orchidaceae2.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p>Most tropical species are epiphytic, often growing on the trunks of trees.</p>
<p align="center">
<img src="media/Orchidaceae3.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p>These epiphytes often have aerial roots with a modified spongy epidermis called a velamen (those of you that took BIOL 210 may remember the beautiful cross-sections you made of these). Here’s a good example:</p>
<p align="center">
<img src="media/Orchidaceae4.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="flowers-2" class="section level2">
<h2>Flowers</h2>
<p>Now this is where orchids really shine! Orchid flowers are truly unique among angiosperms. Their flowers are highly zygomorphic, with 3 sepals and 3 petals. The lower-most petal is modified into a lip or “labellum”, which can take on a huge variety of different forms. Opposite the labellum is the “column”, which is a single fused structure that houses the style, stigma, and stamens. Many of these flowers are the result of very close co-evolution with specific pollinator species. Their magnificent beauty, diversity, and evolutionary history make orchids favourites among horticulturalists and taxonomists alike.</p>
<p>Here are some examples of majestic orchids! Pay close attention to the diversity of flower shapes and sizes, particularly the labellum.</p>
<p align="center">
<img src="media/Orchidaceae5.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Orchidaceae6.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Orchidaceae7.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Orchidaceae8.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Orchidaceae9.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Orchidaceae10.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p align="center">
<img src="media/Orchidaceae11.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<p>Not all orchids have sensationally huge flowers, though. Here are some local examples that are a little more subdued:</p>
<em>Corallorhiza mertensiana</em>
<p align="center">
<img src="media/Orchidaceae14.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<em>Platanthera orbiculata</em>
<p align="center">
<img src="media/Orchidaceae15.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<em>Platanthera dilatata</em>
<p align="center">
<img src="media/Orchidaceae16.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<em>Calypso bulbosa</em> (ok…this one’s actually darling)
<p align="center">
<img src="media/Orchidaceae17.jpeg" style="width:100.0%" />
</p>
<p><br></p>
<hr />
<p>Now let’s take another look at the column, which is less dramatic than the labellum but is much more bizarre when you try to compare it to the reproductive structures of “normal” flowers. At the tip of the column is a structure that represents a single anther. There is an anther cap that is easily pulled off by an exiting pollinator, revealing fairly large yellow structures. These are pollen grains adhering in hard masses called “pollinia” (singular pollinium). The pollinia are attached to a flat, very sticky structure called a “viscidium”, which easily glues the pollinia onto visiting insect pollinators. Upon visiting the next flower, the pollinator will inadvertently stick the pollinia onto the stigmatic surface, which appears as a concave (usually sticky) surface just below the anther cap. There is a small lip between the anther cap and the stigma called a “rostellum”.</p>
<p>See if you can identify all of the structures described above in the footage of the <em>Phalaenopsis</em> orchid below:</p>
<p align="center">
<video src="./media/Orchid_column.mp4" style="width:100.0%" controls=""><a href="./media/Orchid_column.mp4">Video</a></video>
</p>
<p><br></p>
<p>Another interesting thing about orchid flowers is that they are technically “upside down”. The flowers are “resupinate”, meaning they twist 180° during development so that the lower petal is actually the upper petal. It is not super obvious in the photo below, but you can creases in the inferior ovary that indicate twisting:</p>
<p align="center">
<img src="media/Orchidaceae12.jpeg" style="width:49.0%" /> <img src="media/Orchidaceae13.jpeg" style="width:49.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="fruit-1" class="section level2">
<h2>Fruit</h2>
<p>Orchids produce capsules as their fruit type. These tend to be rather unremarkable in comparison to their flowers. Vanilla “beans” are actually capsules, which are full of an oily liquid and thousands (millions?) of tiny, dust-like seeds.</p>
<p align="center">
<img src="media/Orchidaceae18.jpg" style="width:40.0%" />
</p>
<p><br></p>
<hr />
</div>
<div id="knowledge-check-9.3" class="section level2">
<h2>KNOWLEDGE CHECK 9.3</h2>
<p>Choose one of the beautiful orchids pictured above (or pick one of your favourite species ;-)), and make a labeled drawing indicating all visible flower parts. Where possible, make sure to label:</p>
<ul>
<li>sepals</li>
<li>petals</li>
<li>labellum</li>
<li>column
<ul>
<li>label the anther cap, which should be visible.</li>
</ul></li>
</ul>
<hr />
</div>
</div>
<div id="phylogenetics-in-r" class="section level1">
<h1>Phylogenetics in R</h1>
<div id="motivation" class="section level2">
<h2>Motivation</h2>
<p>Phylogenetics is one of the most important and controversial fields within plant taxonomy. The ability to sequence and compare plant DNA has split, grouped, and uncovered taxa in ways never expected. In modern plant taxonomy, phylogenetic relationships are a key piece of evidence to consider when delimiting species. In this lab, we’ll be using R to download and align DNA sequences from five species of <em>Opuntia</em>. Using these sequences, we’ll construct a phylogenetic tree and consider how the species relationships hold up to the published literature.</p>
<hr />
<!-- &nbsp; -->
<!-- &nbsp; -->
<!-- ## Prerequisites  -->
<!-- Before starting this week's tutorial you will have: <br> -->
<!-- 1. completed some pre-assignment reading on DNA barcoding, multiple sequence alignment, phylogenetics, and their role in plant taxonomy, <br> -->
<!-- 2. completed some pre-assignment reading on why/how Genbank is used by plant taxonomists, <br> -->
<!-- 3. navigated through the Genbank website and understand what it is and why it exists.  -->
<!-- 4. **MB** Get students to read (for example, something like) [Majure et al (2012), Am. J. Bot.](https://bsapubs.onlinelibrary.wiley.com/doi/full/10.3732/ajb.1100375) if we end up using _Opuntia_ as an example... -->
<!-- --- -->
<!-- &nbsp; -->
<!-- &nbsp; -->
<!-- #### Outcomes -->
<!-- By the end of this tutorial you will have:  -->
<!-- 1. experience using Genbank, CLUSTAL, and the most common phylogenetics packages in R  -->
<!-- 2. familiarity with Genbank and the types of data that can be obtained,  -->
<!-- 3. the tools and knowledge needed to create your own phylogenetic tree as part of a taxonomic research project,  -->
<!-- --- -->
</div>
<div id="activity-1-downloading-aligning-and-graphing-dna-sequences-from-genbank" class="section level2">
<h2>Activity 1: Downloading, Aligning, and Graphing DNA Sequences from Genbank</h2>
<p><strong>NOTE: You will need R version 4.0 or later. Check your R version by typing in <code>R.Version()</code> into the console.</strong></p>
<p>In previous Assignments we familiarized ourselves with GBIF, R, and the R packages that allow us to connect to databases. Today, we’ll learn how to use our newfound R skills to download and align nucleotide sequences from <a href="https://www.ncbi.nlm.nih.gov/nuccore">GenBank</a>, a repository of DNA sequences hosted by the U.S. National Center for Biotechnology Information (NCBI). We’ll then use the aligned sequences to create a phylogenetic tree that will organize species based on their relatedness.</p>
<p>In this lab, we’ll explore further the genus <em>Opuntia</em>:</p>
<p><em>Opuntia basilaris</em> (Mojave Desert, Colorado, Utah) <em>Opuntia fragilis</em> (Northwestern) <em>Opuntia humifusa</em> (Eastern) <em>Opuntia polyacantha</em> (Great Plains, foothills of Rocky Mountains) <em>Opuntia stricta</em> (Gulf Coast and Caribbean)</p>
<p>As usual, we’ll start by installing the necessary packages:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&#39;ape&#39;</span>)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">install.packages</span>(<span class="st">&#39;seqinr&#39;</span>)</span></code></pre></div>
<p>We’ll install the multiple sequence aligner <code>msa</code> separately:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb2-2"><a href="#cb2-2"></a>    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;msa&quot;</span>)</span></code></pre></div>
<p>At some point the console will ask <code>Update all/some/none? [a/s/n]:</code>. Type the letter <code>n</code> into the console to respond.</p>
<p>Once the packages are installed, we’ll attach them: <code>ape</code> is used to connect with the Genbank database, <code>tidyverse</code> helps us filter and organize our sequence data, <code>msa</code> will align our sequences, <code>seqinr</code> will allow us to concatenate sequences, and we’ll use <code>ape</code> again to construct phylogenetic trees:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># attach packages once they are installed </span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">library</span>(ape)      </span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="kw">library</span>(msa)    </span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="kw">library</span>(seqinr) </span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
<p>First, we’re going to build a <code>data.frame</code> (i.e. a spreadsheet) that will help us keep track of our <em>Opuntia</em> species and the GenBank accession numbers for the <em>rbcL</em> gene for each species. The accession numbers were found on the GenBank website, using the same protocol from Lab 5.</p>
<p><img src="media/lab5_genbank1.png" /></p>
<p>Ahh.. simpler days.</p>
<p>Once the accession numbers are found, they can be saved in a data frame. Make sure to keep your species and accession numbers in the same order!</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># build data frame</span></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># accession numbers found by searching GenBank</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>species &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;O. basilaris&#39;</span>, <span class="st">&#39;O. fragilis&#39;</span>, <span class="st">&#39;O. humifusa&#39;</span>, <span class="st">&#39;O. polyacantha&#39;</span>, <span class="st">&#39;O. stricta&#39;</span>) </span>
<span id="cb4-4"><a href="#cb4-4"></a>rbcL_acc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;JF787189.1&#39;</span>, <span class="st">&#39;JF787223.1&#39;</span>, <span class="st">&#39;JF787227.1&#39;</span>, <span class="st">&#39;JF787259.1&#39;</span>, <span class="st">&#39;JF787288.1&#39;</span> ) </span>
<span id="cb4-5"><a href="#cb4-5"></a></span>
<span id="cb4-6"><a href="#cb4-6"></a>df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(species, rbcL_acc)</span></code></pre></div>
<!-- Now we're ready to search and download the matK sequence for each _Opuntia_ species, using the `read.Genbank()` function: -->
<!-- ```{r} -->
<!-- # call GenBank -->
<!-- matK <-  -->
<!--   read.GenBank( -->
<!--     df$matK_acc, -->
<!--     species.names = FALSE, -->
<!--     as.character = TRUE) %>%  -->
<!--   lapply(., paste0,collapse="") %>%  # remove spaces -->
<!--   lapply(., str_to_upper) # capitalize  -->
<!-- # replace Accession IDs with English names -->
<!-- names(matK) <- df$species -->
<!-- ``` -->
<!-- We can inspect the DNA sequence for _O. basilaris_ by running: -->
<!-- ```{r} -->
<!-- head(matK$`O. basilaris`) -->
<!-- ``` -->
<p>Now we’re ready to search and download the <em>rbcL</em> sequence for each <em>Opuntia</em> species, using the <code>read.Genbank()</code> function:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># call GenBank</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>rbcL &lt;-<span class="st"> </span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="st">  </span><span class="kw">read.GenBank</span>(</span>
<span id="cb5-4"><a href="#cb5-4"></a>    df<span class="op">$</span>rbcL_acc,</span>
<span id="cb5-5"><a href="#cb5-5"></a>    <span class="dt">species.names =</span> <span class="ot">FALSE</span>,</span>
<span id="cb5-6"><a href="#cb5-6"></a>    <span class="dt">as.character =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="st">  </span><span class="kw">lapply</span>(., paste0,<span class="dt">collapse=</span><span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span><span class="co"># remove spaces</span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="st">  </span><span class="kw">lapply</span>(., str_to_upper) <span class="co"># capitalize </span></span>
<span id="cb5-9"><a href="#cb5-9"></a></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co"># replace Accession IDs with English names</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="kw">names</span>(rbcL) &lt;-<span class="st"> </span>df<span class="op">$</span>species</span></code></pre></div>
<p>Cool, <a href="https://pubmed.ncbi.nlm.nih.gov/16100764/">it’s like we’re on CSI</a>.</p>
<p>We now have a ‘DNA sequence object’ named <code>rbcL</code>. It contains the <em>rbcL</em> sequences from each of our five <em>Opuntia</em> species. In the next step, we’ll use the <code>msa()</code> function to align our five <em>rbcL</em> sequences so that their ends are lined up. You could also do this by eye, but it’s a bit tedious and not as reproducible. This is analogous to when we used MUSCLE to align our sequences in Lab 5…</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>rbcL_alignment &lt;-<span class="st"> </span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="st">  </span><span class="kw">msa</span>(<span class="kw">unlist</span>(rbcL),</span>
<span id="cb6-3"><a href="#cb6-3"></a>      <span class="dt">type =</span><span class="st">&#39;dna&#39;</span>, </span>
<span id="cb6-4"><a href="#cb6-4"></a>      <span class="dt">method =</span><span class="st">&#39;ClustalW&#39;</span>,</span>
<span id="cb6-5"><a href="#cb6-5"></a>      <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## use default substitution matrix</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># inspect alignment</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw">print</span>(rbcL_alignment)</span></code></pre></div>
<pre><code>## CLUSTAL 2.1  
## 
## Call:
##    msa(unlist(rbcL), type = &quot;dna&quot;, method = &quot;ClustalW&quot;, verbose = TRUE)
## 
## MsaDNAMultipleAlignment with 5 rows and 807 columns
##     aln                                                    names
## [1] TTTTATAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAATTAAGAG- O. basilaris
## [2] --------------------------...------------------------- O. stricta
## [3] ---------------TATGTCGAAAT...------------------------- O. humifusa
## [4] -----TAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAAT------- O. fragilis
## [5] --TTTTAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAATTAAGTTG O. polyacantha
## Con -----TAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAAT------- Consensus</code></pre>
<!-- ```{r, results = 'hide', message = FALSE} -->
<!-- # matK alignment -->
<!-- matK_alignment <-  -->
<!--   msa(unlist(matK), -->
<!--       type ='dna',  -->
<!--       method ='ClustalW', -->
<!--       verbose = TRUE) -->
<!-- ``` -->
<!-- # ```{r} -->
<!-- # # inspect alignment -->
<!-- # print(matK_alignment)  -->
<!-- # ``` -->
<p>By using the <code>print()</code> function, we see that our sequences are aligned pretty well! You can also notice some differences in nucleotide position between the species. Those are ‘single nucleotide polymorphisms’ (SNPs) that distinguish one species from another.</p>
<p>We’re almost ready to create some phylogenetic trees! First, we’ll convert our alignment objects to <code>seqinr</code> objects. Then, we’ll use the <code>dist.alignment()</code> function to calculate how similar or different each species’ sequence is from one another.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># convert alignment object into seqinr object </span></span>
<span id="cb10-2"><a href="#cb10-2"></a>rbcL_r &lt;-<span class="st"> </span><span class="kw">msaConvert</span>(rbcL_alignment, <span class="dt">type=</span><span class="st">&quot;seqinr::alignment&quot;</span>) </span>
<span id="cb10-3"><a href="#cb10-3"></a>  </span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="co"># create distance matrix</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>rbcL_dist &lt;-<span class="st"> </span><span class="kw">dist.alignment</span>(rbcL_r)</span></code></pre></div>
<p>Now, we’ll use the <code>nj()</code> function to construct trees.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># neighbour joining trees</span></span>
<span id="cb11-2"><a href="#cb11-2"></a>rbcL_tree &lt;-<span class="st"> </span><span class="kw">nj</span>(rbcL_dist)</span></code></pre></div>
<p>Plot your new tree. You can change the layout of the tree by changing the <code>type=</code> argument to any of the following: “phylogram”, “cladogram”, “fan”, or “radial”.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># plot tree</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="kw">plot</span>(rbcL_tree, <span class="dt">main =</span> <span class="st">&#39;rbcL tree&#39;</span>, <span class="dt">type=</span><span class="st">&quot;unrooted&quot;</span>)</span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="kw">add.scale.bar</span>()</span></code></pre></div>
<p><img src="BIOL_324_Lab_9_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p><strong>IMPORTANT</strong> Save your iNaturalist tree as an .rds file so that we can use it in Lab 10. Modify the <code>file =</code> argument if you’d like to save it outside of your RStudio working directory.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw">saveRDS</span>(rbcL_tree, <span class="dt">file =</span> <span class="st">&#39;rbcL_tree.rds&#39;</span>)</span></code></pre></div>
<hr />
</div>
<div id="knowledge-check-9.4" class="section level2">
<h2>KNOWLEDGE CHECK 9.4</h2>
<p>Modify the above code to make a phylogenetic tree of your 5 iNaturalist species instead of <em>Opuntia</em>. Make sure to pick the same 5 species that you chose to map in Lab 7 and save your tree file somewhere on your computer where you can find it again! You’ll be using both your map data from Lab 7 and your tree data from this lab to build your phylogeographic maps in Lab 10 next week - these will be incorporated into Project 2.</p>
<p>Speaking of <a href="BIOL_324_Lab_7.Rmd">Lab 7</a>, we’ve modified the code (see the <code>geometry =</code> argument in the <code>spocc::occ</code> function) to help make your maps more representative of their North American distributions. We encourage you to make this small modification and re-run your Lab 7 scripts so that you can make better maps for Project 2!</p>
<hr />
<!-- For consensus trees: -->
<!-- Recall that matK and rbcL are two separate segments of the chloroplast genome. By sticking them together, we'll create a super-sequence that contains more information than either of the sequences alone. The more sequence information, the better, because it gives us a bigger sample of the entire genome. Concatenation is done by the `seqinr` package, so we'll need to convert our alignment objects into `seqinr` objects: -->
<!-- ```{r, results = 'hide'} -->
<!-- # convert alignment object into seqinr object -->
<!-- matK_r <- msaConvert(matK_alignment, type="seqinr::alignment") -->
<!-- rbcL_r <- msaConvert(rbcL_alignment, type="seqinr::alignment") -->
<!-- # concatenate (merge) sequences -->
<!-- conseq <- mapply(paste0, matK_r$seq, rbcL_r$seq) -->
<!-- names(conseq) <- df$species -->
<!-- # re-align -->
<!-- con_alignment <- -->
<!--   msa(unlist(conseq), -->
<!--       type ='dna', -->
<!--       method ='ClustalW', -->
<!--       verbose = TRUE) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # inspect -->
<!-- print(con_alignment) -->
<!-- # convert alignment object into seqinr object -->
<!-- con_r <- msaConvert(con_alignment, type="seqinr::alignment") -->
<!-- ``` -->
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
