<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mannfred Boehm" />


<title>R Assignment 4: Creating Phylogenetic Trees using Genbank data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOL 324 Labs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="BIOL_324_Lab_6.html">Lab 6</a>
</li>
<li>
  <a href="BIOL_324_Lab_7.html">Lab 7</a>
</li>
<li>
  <a href="BIOL_324_Lab_8.html">Lab 8</a>
</li>
<li>
  <a href="BIOL_324_Lab_9.html">Lab 9</a>
</li>
<li>
  <a href="BIOL_324_Lab_10.html">Lab 10</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Assignment 4: Creating Phylogenetic Trees using Genbank data</h1>
<h4 class="author">Mannfred Boehm</h4>
<h4 class="date">03 August, 2020</h4>

</div>


<p><br> <br></p>
<div id="motivation" class="section level4">
<h4>Motivation</h4>
<p>Add section…</p>
<hr />
<p><br> <br></p>
</div>
<div id="prerequisites" class="section level4">
<h4>Prerequisites</h4>
<p>Before starting this week’s tutorial you will have: <br></p>
<ol style="list-style-type: decimal">
<li><p>completed some pre-assignment reading on DNA barcoding, multiple sequence alignment, phylogenetics, and their role in plant taxonomy, <br></p></li>
<li><p>completed some pre-assignment reading on why/how Genbank is used by plant taxonomists, <br></p></li>
<li><p>navigated through the Genbank website and understand what it is and why it exists.</p></li>
<li><p><strong>MB</strong> Get students to read (for example, something like) Majure et al (2012), Am. J. Bot. if we end up using <em>Opuntia</em> as an example…</p></li>
</ol>
<hr />
<p><br> <br></p>
</div>
<div id="outcomes" class="section level4">
<h4>Outcomes</h4>
<p>By the end of this tutorial you will have: <br></p>
<ol style="list-style-type: decimal">
<li><p>experience using Genbank, CLUSTAL, and the most common phylogenetics packages in R <br></p></li>
<li><p>familiarity with Genbank and the types of data that can be obtained, <br></p></li>
<li><p>the tools and knowledge needed to create your own phylogenetic tree as part of a taxonomic research programme, <br></p></li>
<li><p>…more</p></li>
</ol>
<hr />
<p><br> <br></p>
</div>
<div id="activity-1-downloading-aligning-and-graphing-dna-sequences-from-genbank" class="section level4">
<h4>Activity 1: Downloading, Aligning, and Graphing DNA Sequences from Genbank</h4>
<p>In previous Assignments we familiarized ourselves with GBIF, R, RStudio Cloud, and the R packages that allow us to connect to databases. Today, we’ll learn how to use our newfound R skills to download and align nucleotide sequences from <a href="https://www.ncbi.nlm.nih.gov/nuccore">GenBank</a>, a repository of DNA sequences hosted by the National Center for Biotechnology Information (NCBI). We’ll then use the aligned sequences to create a phylogenetic tree that will organize species based on their relatedness.</p>
<p>In this lab, we’ll explore further the genus <em>Opuntia</em>: <br></p>
<p><em>Opuntia basilaris</em> (Mojave, Colorado, Utah) <br> <em>Opuntia fragilis</em> (Northwestern) <br> <em>Opuntia humifusa</em> (Eastern) <br> <em>Opuntia oricola</em> (Coastal sage and chaparral of S. California) <br> <em>Opuntia polyacantha</em> (Great Plains, foothills of Rocky Mtns) <br></p>
<p><strong>MB</strong> Could add in some natural history background to better motivate this activity. <br></p>
<p>Open this week’s .R file (‘324_Phylogenetics.R’) and navigate to the source pane. Install and attach the necessary packages using <code>install.packages()</code> and <code>library()</code>. By highlighting and running lines 38-111, we can run the script that searches Genbank for <em>matK</em> and <em>rbcL</em> sequences from our five <em>Opuntia</em> species.</p>
<p>Selecting and running lines 120-167 will generate three phylogenetic trees. The first tree illustrates the relationships between species based only on the nucleotide sequences of the <em>matK</em> gene. The second tree does the same thing, but for the <em>rbcL</em> gene. The third tree considers both sequences. Since this tree has the most information, it is often considered a better estimate of the true relationships between species.</p>
<div class="figure">
<img src="matK_tree.jpg" alt="Figure 1: Phylogenetic tree for Opuntia constructed from aligned matK sequences." />
<p class="caption">Figure 1: Phylogenetic tree for <em>Opuntia</em> constructed from aligned <em>matK</em> sequences.</p>
</div>
<p><br></p>
<div class="figure">
<img src="rbcL_tree.jpg" alt="Figure 2: Phylogenetic tree for Opuntia constructed from aligned rbcL sequences." />
<p class="caption">Figure 2: Phylogenetic tree for <em>Opuntia</em> constructed from aligned <em>rbcL</em> sequences.</p>
</div>
<p><strong>MB</strong> Could add in some discussion here about why different genes produce different trees.</p>
<hr />
<p><br> <br></p>
</div>
<div id="activity-2-deciphering-some-code" class="section level4">
<h4>Activity 2: Deciphering some code</h4>
<p>Let’s step through the script in order to understand what is happening here.</p>
<p><br></p>
<p><strong>Lines 14-18</strong>: The <code>library()</code> function loads various packages designed specifically for connecting with the Genbank database (<code>ape</code>), filtering and organizing our data (<code>tidyverse</code>), aligning sequences (<code>msa</code>), concatenating sequences (<code>seqinr</code>), and constructing phylogenetic trees (<code>ape</code>).</p>
<p><strong>Lines 38-40</strong>: The <code>c()</code> concatenates the elements specified inside the brackets <code>()</code> into an object. For example, the <code>species</code> object contains five elements - each one is the name of an <em>Opuntia</em> species that we are interested in. The <code>matK_acc</code> object contains the Genbank accession numbers for each species. These were found by searching Genbank like so:</p>
<div class="figure">
<img src="genbank.jpg" alt="Figure 3: Searching Genbank for accession numbers." />
<p class="caption">Figure 3: Searching Genbank for accession numbers.</p>
</div>
<p><br></p>
<p><strong>Line 42</strong>: From our three objects created above, we will merge them together into a dataframe. The purpose of this dataframe is to keep our accession numbers and species names organized - it will come in handy later!</p>
<p>function searches GBIF for taxon IDs associated with your search query. The <code>&lt;-</code> function stores the results in an object that we named <code>group1_ids</code>. By storing the results in an object we can later return to the object and inspect it, subset it, and feed it to other functions.</p>
<p><strong>Lines 50-56</strong>: The <code>read.Genbank()</code> function searches Genbank for nucleotide sequences associated with our search query. In this case, we are giving <code>read.Genbank()</code> our accession numbers for the <em>matK</em> that we collected earlier.</p>
<p><strong>Line 59</strong>: Because Genbank returns the nucleotide sequences with accession numbers, we want to re-name them with English names for easier reading later on.</p>
<p><strong>Lines 70-79</strong>: Here, we’re collecting sequence information from Genbank, but for the <em>rbcL</em> gene.</p>
<p><strong>Lines 93-98</strong>: Now that we have some DNA sequences, we need to align them. The <code>msa()</code> function takes our sequences and uses the CLUSTAL algorithm to make an optimal alignment (you could also do this by eye, but this becomes difficult with many sequences).</p>
<p><strong>Line 100</strong>: The <code>print()</code> function allows us to inspect the alignment.</p>
<p><strong>Lines 103-111</strong>: Here, we’ll go through the same alignment procedure for the <em>rbcL</em> sequences.</p>
<p><strong>Line 124</strong>: In addition to constructing phylogenies for each gene separately, we’ll also construct a phylogeny using both sequences concatenated together. The <code>paste0()</code> function allows us to string the <em>matK</em> and <em>rbcL</em> sequences together.</p>
<p><strong>Lines 128-135</strong>: With our concatenated sequences, we’ll re-align with CLUSTAL once more using the <code>msa()</code> function.</p>
<p><strong>Lines 148-150</strong>: To construct a phylogeny, we first need to know how similar or dissimlar each species’ DNA sequence is from another. The <code>dist.alignment()</code> function calculates a ‘distance matrix’ that describes these similarities.</p>
<p><strong>Lines 154-156</strong>: With our distance matrices, we can now construct a phylogenetic tree. The <code>nj()</code> function creates a certain type of tree called a ’neighbour joining tree`</p>
<p><strong>MB</strong>: Perhaps need to explain this terminology (distance matrix, neighbour joining) in pre-assingment reading, or lectures.</p>
<p><strong>Line 159-161</strong>: Now it’s time to plot our trees! The <em>matK</em> and <em>rbcL</em> trees were shown earlier, below is the consensus tree:</p>
<div class="figure">
<img src="consensus_tree.jpg" alt="Figure 4: Phylogenetic tree for Opuntia constructed from concatenated and aligned matK and rbcL sequences." />
<p class="caption">Figure 4: Phylogenetic tree for <em>Opuntia</em> constructed from concatenated and aligned <em>matK</em> and <em>rbcL</em> sequences.</p>
</div>
<p><br> <br></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
