<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mannfred Boehm" />


<title>R Assignment 4: Creating Phylogenetic Trees using Genbank data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOL 324 Labs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="BIOL_324_Lab_6.html">Lab 6</a>
</li>
<li>
  <a href="BIOL_324_Lab_7.html">Lab 7</a>
</li>
<li>
  <a href="BIOL_324_Lab_8.html">Lab 8</a>
</li>
<li>
  <a href="BIOL_324_Lab_9.html">Lab 9</a>
</li>
<li>
  <a href="BIOL_324_Lab_10.html">Lab 10</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Assignment 4: Creating Phylogenetic Trees using Genbank data</h1>
<h4 class="author">Mannfred Boehm</h4>
<<<<<<< HEAD
<h4 class="date">10 August, 2020</h4>
=======
<h4 class="date">06 August, 2020</h4>
>>>>>>> aa9c6a795714ba34e7e85a577c4cc343840ae1f1

</div>


<p><br> <br></p>
<div id="motivation" class="section level4">
<h4>Motivation</h4>
<p>Phylogenetics is one of the most important and controversial fields within plant taxonomy. The ability to sequence and compare plant DNA has split, grouped, and uncovered taxa in ways never expected. In modern plant taxonomy, phylogenetic relationships are a key piece of evidence to consider when delimiting species. In this lab, we’ll be downloading and aligning DNA sequences from five species of <em>Opuntia</em>. Using these sequences, we’ll construct a phylogenetic tree and consider how the species relationships hold up to the published literature.</p>
<hr />
<p><br> <br></p>
</div>
<div id="prerequisites" class="section level4">
<h4>Prerequisites</h4>
<p>Before starting this week’s tutorial you will have: <br></p>
<ol style="list-style-type: decimal">
<li><p>completed some pre-assignment reading on DNA barcoding, multiple sequence alignment, phylogenetics, and their role in plant taxonomy, <br></p></li>
<li><p>completed some pre-assignment reading on why/how Genbank is used by plant taxonomists, <br></p></li>
<li><p>navigated through the Genbank website and understand what it is and why it exists.</p></li>
<li><p><strong>MB</strong> Get students to read (for example, something like) Majure et al (2012), Am. J. Bot. if we end up using <em>Opuntia</em> as an example…</p></li>
</ol>
<hr />
<p><br> <br></p>
</div>
<div id="outcomes" class="section level4">
<h4>Outcomes</h4>
<p>By the end of this tutorial you will have:</p>
<ol style="list-style-type: decimal">
<li><p>experience using Genbank, CLUSTAL, and the most common phylogenetics packages in R</p></li>
<li><p>familiarity with Genbank and the types of data that can be obtained,</p></li>
<li><p>the tools and knowledge needed to create your own phylogenetic tree as part of a taxonomic research project,</p></li>
</ol>
<p>   </p>
<hr />
</div>
<div id="activity-1-downloading-aligning-and-graphing-dna-sequences-from-genbank" class="section level4">
<h4>Activity 1: Downloading, Aligning, and Graphing DNA Sequences from Genbank</h4>
<p>In previous Assignments we familiarized ourselves with GBIF, R, and the R packages that allow us to connect to databases. Today, we’ll learn how to use our newfound R skills to download and align nucleotide sequences from <a href="https://www.ncbi.nlm.nih.gov/nuccore">GenBank</a>, a repository of DNA sequences hosted by the U.S. National Center for Biotechnology Information (NCBI). We’ll then use the aligned sequences to create a phylogenetic tree that will organize species based on their relatedness.</p>
<p>In this lab, we’ll explore further the genus <em>Opuntia</em>:</p>
<p><em>Opuntia basilaris</em> (Mojave Desert, Colorado, Utah) <em>Opuntia fragilis</em> (Northwestern) <em>Opuntia humifusa</em> (Eastern) <em>Opuntia oricola</em> (Coastal sage and chaparral of S. California) <em>Opuntia polyacantha</em> (Great Plains, foothills of Rocky Mountains)</p>
<p><strong>MB</strong> Could add in some natural history background to better motivate this activity. <br></p>
<<<<<<< HEAD
<p>As usual, we’ll start by installing the necessary packages using <code>install.packages()</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">install.packages</span>(<span class="kw">c</span>(<span class="st">&#39;ape&#39;</span>, <span class="st">&#39;msa&#39;</span>, <span class="st">&#39;seqinr&#39;</span>))</a></code></pre></div>
<p>Then we’ll attach some packages designed specifically for connecting with the Genbank database (<code>ape</code>), filtering and organizing our data (<code>tidyverse</code>), aligning sequences (<code>msa</code>), concatenating sequences (<code>seqinr</code>), and constructing phylogenetic trees (<code>ape</code>):</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="co"># load packages once they are installed </span></a>
<a class="sourceLine" id="cb2-2" title="2"><span class="kw">library</span>(ape)</a>
<a class="sourceLine" id="cb2-3" title="3"><span class="kw">library</span>(here)</a></code></pre></div>
<pre><code>## here() starts at C:/Users/mannfred/Google Drive/UBC Botany/Archive/UBC_Course_Work/PlantTaxonomy</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="kw">library</span>(msa)</a></code></pre></div>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:grDevices&#39;:
## 
##     windows</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ape&#39;:
## 
##     complement</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" title="1"><span class="kw">library</span>(seqinr)</a></code></pre></div>
<pre><code>## 
## Attaching package: &#39;seqinr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biostrings&#39;:
## 
##     translate</code></pre>
<pre><code>## The following objects are masked from &#39;package:ape&#39;:
## 
##     as.alignment, consensus</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" title="1"><span class="kw">library</span>(tidyverse)</a></code></pre></div>
<pre><code>## -- Attaching packages --------------------------------------- tidyverse 1.3.0 --</code></pre>
<pre><code>## v ggplot2 3.3.0     v purrr   0.3.4
## v tibble  3.0.0     v dplyr   0.8.5
## v tidyr   1.0.2     v stringr 1.4.0
## v readr   1.3.1     v forcats 0.5.0</code></pre>
<pre><code>## -- Conflicts ------------------------------------------ tidyverse_conflicts() --
## x dplyr::collapse()   masks Biostrings::collapse(), IRanges::collapse()
## x dplyr::combine()    masks BiocGenerics::combine()
## x purrr::compact()    masks XVector::compact()
## x dplyr::count()      masks seqinr::count()
## x dplyr::desc()       masks IRanges::desc()
## x tidyr::expand()     masks S4Vectors::expand()
## x dplyr::filter()     masks stats::filter()
## x dplyr::first()      masks S4Vectors::first()
## x dplyr::lag()        masks stats::lag()
## x ggplot2::Position() masks BiocGenerics::Position(), base::Position()
## x purrr::reduce()     masks IRanges::reduce()
## x dplyr::rename()     masks S4Vectors::rename()
## x dplyr::slice()      masks XVector::slice(), IRanges::slice()</code></pre>
<p>First, we’re going to build a spreadsheet that will help us keep track of our <em>Opuntia</em> species and the GenBank accession codes for both the matK and rbcL genes for each species. The accession numbers were found by searching the GenBank website like so:</p>
<div class="figure">
<img src="genbank.jpg" alt="Figure 1: Searching Genbank for accession numbers." />
<p class="caption">Figure 1: Searching Genbank for accession numbers.</p>
</div>
<p>Once the accession numbers are found, they can be saved in a data frame:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" title="1"><span class="co"># build data frame</span></a>
<a class="sourceLine" id="cb31-2" title="2"><span class="co"># accession numbers found by searching GenBank</span></a>
<a class="sourceLine" id="cb31-3" title="3">species &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;O. basilaris&#39;</span>, <span class="st">&#39;O. fragilis&#39;</span>, <span class="st">&#39;O. humifusa&#39;</span>,  <span class="st">&#39;O. oricola&#39;</span>, <span class="st">&#39;O. polyacantha&#39;</span>) </a>
<a class="sourceLine" id="cb31-4" title="4">matK_acc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;JF786750.1&#39;</span>, <span class="st">&#39;JF786786.1&#39;</span>, <span class="st">&#39;JF786791.1&#39;</span>, <span class="st">&#39;JF786812.1&#39;</span>, <span class="st">&#39;JF786823.1&#39;</span>) </a>
<a class="sourceLine" id="cb31-5" title="5">rbcL_acc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;JF787189.1&#39;</span>, <span class="st">&#39;JF787223.1&#39;</span>, <span class="st">&#39;JF787227.1&#39;</span>, <span class="st">&#39;JF787249.1&#39;</span>, <span class="st">&#39;JF787259.1&#39;</span>) </a>
<a class="sourceLine" id="cb31-6" title="6"></a>
<a class="sourceLine" id="cb31-7" title="7">df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(species, matK_acc, rbcL_acc)</a></code></pre></div>
<p>Now we’re ready to search and download the matK sequence for each <em>Opuntia</em> species, using the <code>read.Genbank()</code> function:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" title="1"><span class="co"># call GenBank</span></a>
<a class="sourceLine" id="cb32-2" title="2">matK &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb32-3" title="3"><span class="st">  </span><span class="kw">read.GenBank</span>(</a>
<a class="sourceLine" id="cb32-4" title="4">    df<span class="op">$</span>matK_acc,</a>
<a class="sourceLine" id="cb32-5" title="5">    <span class="dt">species.names =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb32-6" title="6">    <span class="dt">as.character =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb32-7" title="7"><span class="st">  </span><span class="kw">lapply</span>(., paste0,<span class="dt">collapse=</span><span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span><span class="co"># remove spaces</span></a>
<a class="sourceLine" id="cb32-8" title="8"><span class="st">  </span><span class="kw">lapply</span>(., str_to_upper) <span class="co"># capitalize </span></a>
<a class="sourceLine" id="cb32-9" title="9"></a>
<a class="sourceLine" id="cb32-10" title="10"><span class="co"># replace Accession IDs with English names</span></a>
<a class="sourceLine" id="cb32-11" title="11"><span class="kw">names</span>(matK) &lt;-<span class="st"> </span>df<span class="op">$</span>species</a></code></pre></div>
<p>We can inspect the DNA sequence for <em>O. basilaris</em> by running:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" title="1"><span class="kw">head</span>(matK<span class="op">$</span><span class="st">`</span><span class="dt">O. basilaris</span><span class="st">`</span>)</a></code></pre></div>
<pre><code>## [1] &quot;TCAATATTTCCCTTTTTCGAGGACAAATTCTTACATTTAAATTATGTGTTAAAAATATTAATACCTTACCCCATTCATCTAGAAATCTTGGTTCAAACTCTTCGTTACTGGGTGAAAGATGCTTCTTCTTTGCATTTATTACGATTCTTTCTTTATGAGTATCGTAATTGGAATAGTCTTATTACTCCCCAAAAATCCATTTCTATTTTTTCAAAAAAGAATCAACGATTATTCTTGTTCCTATATAATTTCCATGTATGTGAATACGAATCCATTTTCGTTTTTCTCTGTAACCAATCCTCTCATTTACGATCTTTTACTTTTGGAGCCCTTCTTGAACGAATCTATTTTTATGGAAAGCGTGAATATCTAGTAAAAGTAAAAACTTTATCTAAGGATTTTCGTCTTATCTTATGGCTTTTCAAAGACCCTTTCCTGCATTATGAATTGTATCGAGGAAAAGAAATTCTGGCTTCAAAGGGGACATCTCTTCTGATGCATAAATGGAAATATTATCTTAACAATTTTTGGCAATGCCATTTTTCCCTGTGGTCTCAACCAAGAAGAATCTATATCAATCGATTATCAAAGCATTCTCTCGACTTTATGAGTTTTTTTTCAAGTGTTCGACTCAATTCTTCAGTGGTACGGAGTCAAATGGTAGAAAATTCATTTCTAATAGATAATCCTATTAAGAAATTCGATACCATAGTTCGAATTATTCCTCTGGTTGGATCGTTGGCTAAAGCGAAATTTTGTAACGTATTAGGACATCCCGTTAGTAAGTCGGTCTGGACCGATTTATTAGATTCTGATATTATTGATCGGTTTGGGCGCATATGCAGAAATCTTTCTCATTATTATAGTGGCTCTTCAAGAAAAAAGAGTTTGTATCGAATAAAGTATATA&quot;</code></pre>
<p>Cool, <a href="https://pubmed.ncbi.nlm.nih.gov/16100764/">I feel like I’m on CSI</a>. The code for searching and downloading the rbcL sequences is similar:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" title="1"><span class="co"># call GenBank</span></a>
<a class="sourceLine" id="cb35-2" title="2">rbcL &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb35-3" title="3"><span class="st">  </span><span class="kw">read.GenBank</span>(</a>
<a class="sourceLine" id="cb35-4" title="4">    df<span class="op">$</span>rbcL_acc,</a>
<a class="sourceLine" id="cb35-5" title="5">    <span class="dt">species.names =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb35-6" title="6">    <span class="dt">as.character =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-7" title="7"><span class="st">  </span><span class="kw">lapply</span>(., paste0,<span class="dt">collapse=</span><span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span><span class="co"># remove spaces</span></a>
<a class="sourceLine" id="cb35-8" title="8"><span class="st">  </span><span class="kw">lapply</span>(., str_to_upper) <span class="co"># capitalize </span></a>
<a class="sourceLine" id="cb35-9" title="9"></a>
<a class="sourceLine" id="cb35-10" title="10"><span class="co"># replace Accession IDs with English names</span></a>
<a class="sourceLine" id="cb35-11" title="11"><span class="kw">names</span>(rbcL) &lt;-<span class="st"> </span>df<span class="op">$</span>species</a></code></pre></div>
<p>We now have two DNA sequence objects, <code>matK</code> and <code>rbcL</code>, that contain sequences from each of our five <em>Opuntia</em> species. In the next step, we’ll use the <code>msa()</code> function to align our five matK sequences so that their ends are lined up. You could also do this by eye, but it’s a bit tedious and not as reproducible:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" title="1"><span class="co"># matK alignment</span></a>
<a class="sourceLine" id="cb36-2" title="2">matK_alignment &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb36-3" title="3"><span class="st">  </span><span class="kw">msa</span>(<span class="kw">unlist</span>(matK),</a>
<a class="sourceLine" id="cb36-4" title="4">      <span class="dt">type =</span><span class="st">&#39;dna&#39;</span>, </a>
<a class="sourceLine" id="cb36-5" title="5">      <span class="dt">method =</span><span class="st">&#39;ClustalW&#39;</span>,</a>
<a class="sourceLine" id="cb36-6" title="6">      <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## use default substitution matrix</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" title="1"><span class="co"># inspect alignment</span></a>
<a class="sourceLine" id="cb38-2" title="2"><span class="kw">print</span>(matK_alignment) </a></code></pre></div>
<pre><code>## CLUSTAL 2.1  
## 
## Call:
##    msa(unlist(matK), type = &quot;dna&quot;, method = &quot;ClustalW&quot;, verbose = TRUE)
## 
## MsaDNAMultipleAlignment with 5 rows and 911 columns
##     aln                                                    names
## [1] ----CATATT-CCCTTTT-CGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. fragilis
## [2] ----CATATTTCCCTTTT-CGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. humifusa
## [3] ---TCATATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. oricola
## [4] CATCAATATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. polyacantha
## [5] --TCAATATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA O. basilaris
## Con ---?CATATTTCCCTTTTTCGAGGAC...GAGTTTGTATCGAATAAAGTATATA Consensus</code></pre>
<p>By using the <code>print()</code> function, we see that our sequences are aligned pretty well! You can also notice some differences in nucleotide position between the species. Those are ‘single nucleotide polymorphisms’ (SNPs) that, when considered across the entire genome, distinguish one species from another.</p>
<p>We’ll now align our rbcL sequences:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" title="1">rbcL_alignment &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb40-2" title="2"><span class="st">  </span><span class="kw">msa</span>(<span class="kw">unlist</span>(rbcL),</a>
<a class="sourceLine" id="cb40-3" title="3">      <span class="dt">type =</span><span class="st">&#39;dna&#39;</span>, </a>
<a class="sourceLine" id="cb40-4" title="4">      <span class="dt">method =</span><span class="st">&#39;ClustalW&#39;</span>,</a>
<a class="sourceLine" id="cb40-5" title="5">      <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## use default substitution matrix</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" title="1"><span class="co"># inspect alignment</span></a>
<a class="sourceLine" id="cb42-2" title="2"><span class="kw">print</span>(rbcL_alignment)</a></code></pre></div>
<pre><code>## CLUSTAL 2.1  
## 
## Call:
##    msa(unlist(rbcL), type = &quot;dna&quot;, method = &quot;ClustalW&quot;, verbose = TRUE)
## 
## MsaDNAMultipleAlignment with 5 rows and 807 columns
##     aln                                                    names
## [1] TTTTATAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAATTAAGAG- O. basilaris
## [2] ---------------TATGTCGAAAT...------------------------- O. humifusa
## [3] -----TAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAAT------- O. fragilis
## [4] --TTTTAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAATTAAGTTG O. polyacantha
## [5] --------------------------...------------------------- O. oricola
## Con -----TAAAGTAAACTATGTCGAAAT...TAAAAATTTTCATCAAAT------- Consensus</code></pre>
<p>Recall that matK and rbcL are two separate segments of the chloroplast genome. By sticking them together, we’ll create a super-sequence that contains more information than either of the sequences alone. The more sequence information, the better, because it gives us a bigger sample of the entire genome. Concatenation is done by the <code>seqinr</code> package, so we’ll need to convert our alignment objects into <code>seqinr</code> objects:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" title="1"><span class="co"># convert alignment object into seqinr object</span></a>
<a class="sourceLine" id="cb44-2" title="2">matK_r &lt;-<span class="st"> </span><span class="kw">msaConvert</span>(matK_alignment, <span class="dt">type=</span><span class="st">&quot;seqinr::alignment&quot;</span>)</a>
<a class="sourceLine" id="cb44-3" title="3">rbcL_r &lt;-<span class="st"> </span><span class="kw">msaConvert</span>(rbcL_alignment, <span class="dt">type=</span><span class="st">&quot;seqinr::alignment&quot;</span>)</a>
<a class="sourceLine" id="cb44-4" title="4"></a>
<a class="sourceLine" id="cb44-5" title="5"><span class="co"># concatenate (merge) sequences</span></a>
<a class="sourceLine" id="cb44-6" title="6">conseq &lt;-<span class="st"> </span><span class="kw">mapply</span>(paste0, matK_r<span class="op">$</span>seq, rbcL_r<span class="op">$</span>seq)</a>
<a class="sourceLine" id="cb44-7" title="7"><span class="kw">names</span>(conseq) &lt;-<span class="st"> </span>df<span class="op">$</span>species</a>
<a class="sourceLine" id="cb44-8" title="8"></a>
<a class="sourceLine" id="cb44-9" title="9"><span class="co"># re-align</span></a>
<a class="sourceLine" id="cb44-10" title="10">con_alignment &lt;-</a>
<a class="sourceLine" id="cb44-11" title="11"><span class="st">  </span><span class="kw">msa</span>(<span class="kw">unlist</span>(conseq),</a>
<a class="sourceLine" id="cb44-12" title="12">      <span class="dt">type =</span><span class="st">&#39;dna&#39;</span>, </a>
<a class="sourceLine" id="cb44-13" title="13">      <span class="dt">method =</span><span class="st">&#39;ClustalW&#39;</span>,</a>
<a class="sourceLine" id="cb44-14" title="14">      <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## use default substitution matrix</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" title="1"><span class="co"># inspect </span></a>
<a class="sourceLine" id="cb46-2" title="2"><span class="kw">print</span>(con_alignment)</a></code></pre></div>
<pre><code>## CLUSTAL 2.1  
## 
## Call:
##    msa(unlist(conseq), type = &quot;dna&quot;, method = &quot;ClustalW&quot;, verbose = TRUE)
## 
## MsaDNAMultipleAlignment with 5 rows and 1718 columns
##     aln                                                    names
## [1] ----CATATT-CCCTTTT-CGAGGAC...TAAAAATTTTCATCAAATTAAGAG- O. basilaris
## [2] ----CATATTTCCCTTTT-CGAGGAC...------------------------- O. fragilis
## [3] ---TCATATTTCCCTTTTTCGAGGAC...TAAAAATTTTCATCAAAT------- O. humifusa
## [4] CATCAATATTTCCCTTTTTCGAGGAC...TAAAAATTTTCATCAAATTAAGTTG O. oricola
## [5] --TCAATATTTCCCTTTTTCGAGGAC...------------------------- O. polyacantha
## Con ---?CATATTTCCCTTTTTCGAGGAC...TAAAAATTTTCATCAAAT------- Consensus</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" title="1"><span class="co"># convert alignment object into seqinr object</span></a>
<a class="sourceLine" id="cb48-2" title="2">con_r &lt;-<span class="st"> </span><span class="kw">msaConvert</span>(con_alignment, <span class="dt">type=</span><span class="st">&quot;seqinr::alignment&quot;</span>)</a></code></pre></div>
<p>We’re ready to create some phylogenetic trees! First, we’ll use the <code>dist.alignment()</code> function to calculate how similar or different each species’ sequence is from one another. We’ll do this for the matK, rbcL, and concatenated sequences, so that we can compare the trees:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" title="1"><span class="co"># create distance matrices</span></a>
<a class="sourceLine" id="cb49-2" title="2">matK_dist &lt;-<span class="st"> </span><span class="kw">dist.alignment</span>(matK_r)</a>
<a class="sourceLine" id="cb49-3" title="3">rbcL_dist &lt;-<span class="st"> </span><span class="kw">dist.alignment</span>(rbcL_r)</a>
<a class="sourceLine" id="cb49-4" title="4">con_dist &lt;-<span class="st"> </span><span class="kw">dist.alignment</span>(con_r)</a></code></pre></div>
<p>Now, we’ll use the <code>nj()</code> function to construct trees:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" title="1"><span class="co"># neighbour joining trees</span></a>
<a class="sourceLine" id="cb50-2" title="2">matK_tree &lt;-<span class="st"> </span><span class="kw">nj</span>(matK_dist)</a>
<a class="sourceLine" id="cb50-3" title="3">rbcL_tree &lt;-<span class="st"> </span><span class="kw">nj</span>(rbcL_dist)</a>
<a class="sourceLine" id="cb50-4" title="4">con_tree &lt;-<span class="st"> </span><span class="kw">nj</span>(con_dist)</a></code></pre></div>
<p>Plot your new trees:</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" title="1"><span class="co"># plot trees</span></a>
<a class="sourceLine" id="cb51-2" title="2"><span class="kw">plot</span>(matK_tree, <span class="dt">main =</span><span class="st">&#39;matK tree&#39;</span>)</a></code></pre></div>
<p><img src="BIOL_324_Lab_9_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" title="1"><span class="kw">plot</span>(rbcL_tree, <span class="dt">main =</span> <span class="st">&#39;rbcL tree&#39;</span>)</a></code></pre></div>
<p><img src="BIOL_324_Lab_9_files/figure-html/unnamed-chunk-12-2.png" width="672" /></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" title="1"><span class="kw">plot</span>(con_tree, <span class="dt">main =</span> <span class="st">&#39;Consensus tree (matK + rbcL)&#39;</span>) </a></code></pre></div>
<p><img src="BIOL_324_Lab_9_files/figure-html/unnamed-chunk-12-3.png" width="672" /></p>
<p><strong>-KNOWLEDGE CHECK-</strong> Do the trees differ? Which phylogenetic relationships have been re-arranged between trees?</p>
=======
<p>Open this week’s .R file (‘324_Phylogenetics.R’) and navigate to the source pane. Install and attach the necessary packages using <code>install.packages()</code> and <code>library()</code>. By highlighting and running lines 38-111, we can run the script that searches Genbank for <em>matK</em> and <em>rbcL</em> sequences from our five <em>Opuntia</em> species.</p>
<p>Selecting and running lines 120-167 will generate three phylogenetic trees. The first tree illustrates the relationships between species based only on the nucleotide sequences of the <em>matK</em> gene. The second tree does the same thing, but for the <em>rbcL</em> gene. The third tree considers both sequences. Since this tree has the most information, it is often considered a better estimate of the true relationships between species.</p>
<div class="figure">
<img src="matK_tree.jpg" alt="" />
<p class="caption">Figure 1: Phylogenetic tree for <em>Opuntia</em> constructed from aligned <em>matK</em> sequences.</p>
</div>
<p><br></p>
<div class="figure">
<img src="rbcL_tree.jpg" alt="" />
<p class="caption">Figure 2: Phylogenetic tree for <em>Opuntia</em> constructed from aligned <em>rbcL</em> sequences.</p>
</div>
<p><strong>MB</strong> Could add in some discussion here about why different genes produce different trees.</p>
<hr />
<p><br> <br></p>
</div>
<div id="activity-2-deciphering-some-code" class="section level4">
<h4>Activity 2: Deciphering some code</h4>
<p>Let’s step through the script in order to understand what is happening here.</p>
<p><br></p>
<p><strong>Lines 14-18</strong>: The <code>library()</code> function loads various packages designed specifically for connecting with the Genbank database (<code>ape</code>), filtering and organizing our data (<code>tidyverse</code>), aligning sequences (<code>msa</code>), concatenating sequences (<code>seqinr</code>), and constructing phylogenetic trees (<code>ape</code>).</p>
<p><strong>Lines 38-40</strong>: The <code>c()</code> concatenates the elements specified inside the brackets <code>()</code> into an object. For example, the <code>species</code> object contains five elements - each one is the name of an <em>Opuntia</em> species that we are interested in. The <code>matK_acc</code> object contains the Genbank accession numbers for each species. These were found by searching Genbank like so:</p>
<div class="figure">
<img src="genbank.jpg" alt="" />
<p class="caption">Figure 3: Searching Genbank for accession numbers.</p>
</div>
<p><br></p>
<p><strong>Line 42</strong>: From our three objects created above, we will merge them together into a dataframe. The purpose of this dataframe is to keep our accession numbers and species names organized - it will come in handy later!</p>
<p>function searches GBIF for taxon IDs associated with your search query. The <code>&lt;-</code> function stores the results in an object that we named <code>group1_ids</code>. By storing the results in an object we can later return to the object and inspect it, subset it, and feed it to other functions.</p>
<p><strong>Lines 50-56</strong>: The <code>read.Genbank()</code> function searches Genbank for nucleotide sequences associated with our search query. In this case, we are giving <code>read.Genbank()</code> our accession numbers for the <em>matK</em> that we collected earlier.</p>
<p><strong>Line 59</strong>: Because Genbank returns the nucleotide sequences with accession numbers, we want to re-name them with English names for easier reading later on.</p>
<p><strong>Lines 70-79</strong>: Here, we’re collecting sequence information from Genbank, but for the <em>rbcL</em> gene.</p>
<p><strong>Lines 93-98</strong>: Now that we have some DNA sequences, we need to align them. The <code>msa()</code> function takes our sequences and uses the CLUSTAL algorithm to make an optimal alignment (you could also do this by eye, but this becomes difficult with many sequences).</p>
<p><strong>Line 100</strong>: The <code>print()</code> function allows us to inspect the alignment.</p>
<p><strong>Lines 103-111</strong>: Here, we’ll go through the same alignment procedure for the <em>rbcL</em> sequences.</p>
<p><strong>Line 124</strong>: In addition to constructing phylogenies for each gene separately, we’ll also construct a phylogeny using both sequences concatenated together. The <code>paste0()</code> function allows us to string the <em>matK</em> and <em>rbcL</em> sequences together.</p>
<p><strong>Lines 128-135</strong>: With our concatenated sequences, we’ll re-align with CLUSTAL once more using the <code>msa()</code> function.</p>
<p><strong>Lines 148-150</strong>: To construct a phylogeny, we first need to know how similar or dissimlar each species’ DNA sequence is from another. The <code>dist.alignment()</code> function calculates a ‘distance matrix’ that describes these similarities.</p>
<p><strong>Lines 154-156</strong>: With our distance matrices, we can now construct a phylogenetic tree. The <code>nj()</code> function creates a certain type of tree called a ’neighbour joining tree`</p>
<p><strong>MB</strong>: Perhaps need to explain this terminology (distance matrix, neighbour joining) in pre-assingment reading, or lectures.</p>
<p><strong>Line 159-161</strong>: Now it’s time to plot our trees! The <em>matK</em> and <em>rbcL</em> trees were shown earlier, below is the consensus tree:</p>
<div class="figure">
<img src="consensus_tree.jpg" alt="" />
<p class="caption">Figure 4: Phylogenetic tree for <em>Opuntia</em> constructed from concatenated and aligned <em>matK</em> and <em>rbcL</em> sequences.</p>
</div>
<p><br> <br></p>
>>>>>>> aa9c6a795714ba34e7e85a577c4cc343840ae1f1
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
